### DISCUSSION TITLE
clear:cache from var/cache/common

### DISCUSSION THREAD

**Discussion by rantnik** [sulu/sulu#6153] [Q&A]
Hi guys 👋
Faced with a problem like this https://github.com/sulu/sulu/issues/5423
Can you please tell me the command to delete cache from `var/cache/common` folder? I searched for a solution, but could not find it. `rm -rf` takes a very long time for some reason (I have ~50k files and 100k folders).

**Comment by alexander-schranz** [comment]
Hello @rantnik the `rm -rf` is the most efficient way to remove the http cache cache directory. As shown in the linked issue there are some http cache commands but they are not as efficient as doing it directly on operation system level with `rm -rf`.

If you just wan to to delete the http cache directory you just select the http cache directory not sure should be something like:

```bash
rm -rf var/cache/common/prod/http_cache
```

But I think this is no difference to:

```bash
rm -rf var/cache/
```

As the only thing which is writing a lot of files is the http cache and the node cache. And it should not hurt if you cleanup old files there also.

PS: If you want to clean the http cache on deployment you can also use the following command:

```bash
bin/console fos:httpcache:clear
```

Keep in mind that this should only be done for websites which has not a lot traffic because removing the whole http cache at once could end on a high load on your webserver.

**Reply by rantnik** [reply]
Alexander, thank you very much for your reply.
What could be the reason for the very long deletion of cache files with `rm -rf` command (several minutes and time out, 50k files and 120k subdirectories)?

**Reply by rantnik** [reply]
Could you advise on the best way to clean the cache after updates (`git pull`) on a high-traffic website?

**Reply by alexander-schranz** [reply]
@rantnik On a high traffic website you should not use `git pull` you should use a clean Blue / Green Deployment. The simpliest way here is deployer. Have a look at https://sulu.io/blog/deploy-sulu-with-deployer. Also on a high traffic website you should use `Varnish` as your HttpCache and not the Symfony file based Http Cache.

**Reply by alexander-schranz** [reply]
Varnish has a mode called grace mode which will return always cached pages and refresh them in the background but for this varnish and sulu need to configure correctly to use the `xkey` feature.

**Reply by rantnik** [reply]
@alexander-schranz thank you!
